docs/src/references/mqtt-topic-guidelines.md

```diff
@@ -0,0 +1,165 @@
+
+... where there is inconsistency in the placement of child devices and how the commands are grouped.
+
+There are a few other limitations like the lack of support for services on the thin-edge device,
+difficulty in extending existing topics with additional sub-topics, etc
+which are detailed in the requirements section.
+
+## Requirements
+
+This section is divided into 3 parts:
+1. Must-have: for the requirements that must be met by the proposed solutions
+2. Nice-to-have: these requirements are not mandatory, but the solution addressing more of these requirements would be a plus
+3. Out-of-scope: those that are relevant but out of scope for this design exercise
+
+### Must-have
+
+1. A topic structure to receive telemetry data from services running on the main device,
+   which can't be associated to the main device or some child device,
+   but must be associated to a logical service entity that is linked to the main device.
+1. Consistency in the topic structures for the main device, child device and services
+1. Support nested child devices so that telemetry data and commands can be received from/sent to child devices of child devices.
+   deployments with at least 3 levels of nesting: a thin-edge device, its children and grand-children are not uncommon.
+1. When device IDs are used in topics, it is expected that all nested child devices under a thin-edge device are unique.
+   They need not be globally unique or even unique across multiple thin-edge devices.
+1. If child devices can not ensure uniqueness in their IDs,
+   a registration step with thin-edge can be used to get unique IDs under the thin-edge namespace.
+   This registration step must be optional for devices with unique IDs.
+1. Support the following kinds of filtering with minimal effort (ideally with a single wildcard subscription):
+   * All data from thin-edge device excluding everything else(other services and child devices)
+   * All data from all child devices excluding those from thin-edge device and services
+   * All data from all services excluding those from thin-edge device and child devices
+   * All data from a specific child device excluding everything else
+   * All data from a specific service excluding everything else
+1. The topic structure should be ACL friendly so that rules can be applied
+   to limit child devices and services to access only the data meant for them.
+1. Enable easier extension of topics with further topic suffixes in future:
+   E.g: Support `type` in the topic for measurements like `tedge/measurements/{measurement-type}`
+
+### Nice-to have
+
+1. Dynamic creation/registration of child devices on receipt of the very first data that they send.
+   This is desired at least for immediate child devices, if not for further nested child devices.
+1. Easy to create static routing rules so that it is easy to map a local MQTT topic for a nested child device
+   to the equivalent cloud topic for its twin.
```

Comment:
The downside with flattening the device hierarchy in the topic names (e.g. `left-front-wheel/brake-sensors` => `left-front-wheel-brake-sensors`) is that you can't use the mosquitto bridge for a local broker to broker translation as it requires each topic to end with a `/` and it can only add or subtract a prefix. Using the mosquitto bridge functionality for local bridge  to bridge communication is useful if you have the grand children running in a different network to the parent devices, so you need a bridge between the parents and the grand children (that is where using the in-built mosquitto bridge functionality would be useful because it does not require any additional software to do the mapping, it can be down by the broker directly.